aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/internal/CompleteMultipartUpload.java

```diff
@@ -115,17 +115,23 @@ public UploadResult call() throws Exception {
      */
     private List<PartETag> collectPartETags() {
 
-        final List<PartETag> partETags = new ArrayList<PartETag>();
+        final List<PartETag> partETags = new ArrayList<PartETag>(eTagsBeforeResume.size());
         partETags.addAll(eTagsBeforeResume);
-        for (Future<PartETag> future : futures) {
-            try {
-                partETags.add(future.get());
-            } catch (Exception e) {
-                throw new SdkClientException(
-                        "Unable to complete multi-part upload. Individual part upload failed : "
-                                + e.getCause().getMessage(), e.getCause());
+
+        int index = 0;
+        try {
+            for (; index < futures.size(); index++) {
+                partETags.add(futures.get(index).get());
             }
+        } catch (Exception e) {
+            for (; index < futures.size(); index++) {
```

Comment:
Don't we need to cancel all the uploads? 
Based on this implementation, all the uploads next to the exception-upload will be canceled (including the upload which got exception). So what about the uploads happened before to this exception?